Makefile

```diff
@@ -215,7 +231,8 @@ LIB_SOURCES += utilities/env_librados.cc
 LDFLAGS += -lrados
 endif
 
-AM_LINK = $(AM_V_CCLD)$(CXX) $^ $(EXEC_LDFLAGS) -o $@ $(LDFLAGS) $(COVERAGEFLAGS)
+AM_LINK = $(AM_V_CCLD)$(CXX) $(patsubst lib%.a, -l%, $(patsubst lib%.$(PLATFORM_SHARED_EXT), -l%, $^)) $(EXEC_LDFLAGS) -o $@ -L. $(LDFLAGS) $(COVERAGEFLAGS)
```

Comment:
I think the -L. should be before the libraries (patsubst).

Reply:
It failed for me the way you had it so I semi-reverted. https://github.com/facebook/rocksdb/pull/6660/commits/b94b715e5909a9a8074533dcf13488f2314d84e9

Reply:
But moving just the -L. does seem ok testing it just now. (Now for follow-up PR)